src/seedu/addressbook/storage/StorageFile.java

```diff
@@ -125,29 +100,27 @@ public AddressBook load() throws StorageOperationException {
             return new AddressBook();
         }
 
-        try (final Reader fileReader =
-                     new BufferedReader(new FileReader(path.toFile()))) {
-
-            final Unmarshaller unmarshaller = jaxbContext.createUnmarshaller();
-            final AdaptedAddressBook loaded = (AdaptedAddressBook) unmarshaller.unmarshal(fileReader);
-            // manual check for missing elements
-            if (loaded.isAnyRequiredFieldMissing()) {
-                throw new StorageOperationException("File data missing some elements");
-            }
-            return loaded.toModelType();
-
+        try {
+            return AddressBookDecoder.decodeAddressBook(getLinesInFile(path));
         } catch (FileNotFoundException fnfe) {
             throw new AssertionError("A non-existent file scenario is already handled earlier.");
         // other errors
         } catch (IOException ioe) {
             throw new StorageOperationException("Error writing to file: " + path);
-        } catch (JAXBException jaxbe) {
-            throw new StorageOperationException("Error parsing file data format");
         } catch (IllegalValueException ive) {
             throw new StorageOperationException("File contains illegal data values; data type constraints not met");
         }
     }
 
+    /**
+     * Gets all lines from the file as a list of strings. Line separators are removed.
+     *
+     * @throws IOException if there is an error reading the file.
+     */
+    private static List<String> getLinesInFile(Path dataFilePath) throws IOException {
+        return new ArrayList<>(Files.readAllLines(dataFilePath));
```

Comment:
is there a reason for wrapping it in a new `ArrayList`?

Reply:
Also, I'm wondering whether we can just inline this method.

Reply:
>is there a reason for wrapping it in a new ArrayList?

Not sure, just copied from AB1.

>Also, I'm wondering whether we can just inline this method.

Yep, sounds good.